Fix/series cache language

skaik-mo opened the pull request:
Resolved an issue where series in the cache were not updated correctly after changing the app language. The fix ensures that cached series are cleared and re-synced to reflect the selected language.

skaik-mo committed:
Refactor: Update series genre data handling

This commit refactors the way series genre data is handled:

- Series genres are now observed as a Flow from the local data source.
- `GetSeriesGenresUseCase` is renamed to `ObserveSeriesGenresUseCase` to reflect this change.
- A new `SyncSeriesGenresUseCase` is introduced to synchronize genres from the remote source.
- `GetSeriesGenresByIdsUseCase` now returns a Flow of genres.
- The local data source's `getGenres` and `getGenresByIds` methods now return Flows.
- The `insertGenres` method in the local data source is replaced by `syncGenres` for better clarity.
- Recently viewed series are now added when fetching series details, removing the `AddRecentSeriesUseCase` from `SeriesDetailsViewModel`.
- Series Dao method names for recently viewed series are updated for consistency.

skaik-mo committed:
Merge branch 'fix/movie-cache-language' into fix/series-cache-language

* fix/movie-cache-language: (47 commits)
  Test: Update GetCollectionsUseCaseTest
  rename package category_media to categoryMedia
  fix get collections with no session id
  refactor poster ui in home feature
  Center no results message in MatchResultScreen
  Update `isEmptyResults` state in `MatchResultViewModel`
  Fix full date format to be year first
  rename package name show more to category media
  Standardize progress indicator size in buttons
  Refactor login status checking in RatingViewModel and MyCollectionsViewModel
  refactor view model Category Media
  refactor format duration
  Refactor: Check login status before fetching ratings in RatingViewModel
  Refactor MyCollectionsViewModel to check login status before fetching collections
  Make EncryptionService cipher a local val
  fix poster horizontal in home composable
  fix mapper ui to use format date and duration
  rename show more files and refactor view model
  create extension format date and duration
  refactor no internet in home screen
  ...

# Conflicts:
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/BasedOnTrueEventsStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/CinematicMasterpiecesStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/FamilyNightPicksStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/FeelGoodPreferencesStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/LateNightThrillsStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/MatchesYourVibesStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/MindBendingStoriesStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/RecentlyReleasedStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/RecentlyViewedStrategy.kt

skaik-mo committed:
Merge branch 'fix/movie-cache-language' into fix/series-cache-language

* fix/movie-cache-language: (22 commits)
  Assume user is logged in for GetCollectionsUseCaseTest
  add IsLoggedInUseCase in GetCollectionsUseCase
  delete unused import
  add horizontal slide animation to nav graph
  update animation for bottom bar
  move back handler from main to home nav graph
  fix: animation of show more screen
  Refactor network module to remove RetrofitRequestBuilder and simplify data source implementations
  Refactor MoviesRemoteDataSourceImpl and MoviesApiService
  Refactor MediaMemberApiService and MediaMemberRemoteDataSource
  refactor: rename safeCall to safeCallRemote and MatchRemoteDataSourceImplRetrofit to MatchRemoteDataSourceImpl
  Refactor CollectionsApiService and CollectionsRemoteDataSource
  Refactor SeriesApiService and SeriesRemoteDataSource
  Refactor SearchRemoteDataSourceImpl to use safeCallRemote and rename SearchApiServiceRetrofit
  refactor: move safeCall and mapToMediaException to SafeCallRemote.kt
  fix: animation in explore screen
  Poster List Section Move
  fix movie and series details alignment
  fix cast details animation
  close app on back pressed in main graph
  ...

skaik-mo committed:
Refactor: Enhance Series Cache Management and Popular Series Observation

This commit refactors series cache management and how popular series are observed.

**Key Changes:**

*   **Removed `ClearRecentlyViewedSeriesUseCase`:** Its functionality is now integrated into `ClearSeriesCacheUseCase`.
*   **Enhanced `ClearSeriesCacheUseCase`:**
    *   Added `clearExceptRecentlyViewed()` method to clear all series cache except for recently viewed items.
    *   Added `clearRecentlyViewed()` method to clear only recently viewed series.
    *   The existing `invoke()` method is renamed to `clearAll()` for clarity.
*   **Refactored `GetPopularitySeriesUseCase` to `ObservePopularSeriesUseCase`:**
    *   The use case now returns a `Flow` to observe changes in popular series.
    *   It no longer takes a `page` parameter, relying on the repository to handle pagination if needed.
*   **Updated `SeriesRepository` and `SeriesLocalDataSource`:**
    *   `getPopularitySeries` in `SeriesLocalDataSource` and `observePopular` in `SeriesRepository` now return `Flow<List<SeriesCacheDto>>` and `Flow<List<Series>>` respectively, enabling observation of popular series.
    *   Added `clearExceptRecentlyViewed()` and `clearAll()` methods to both interfaces and their implementations for more granular cache control.
    *   `SeriesDao`'s `getPopularitySeries` now returns a `Flow`.
    *   Added `clearSeriesCache()` method to `SeriesDao` to specifically clear the main series table.
*   **Updated ViewModels and Workers:**
    *   `SettingsViewModel`, `SearchViewModel`, and `CacheCleanupWorker` now use the updated methods in `ClearSeriesCacheUseCase` for cache clearing operations.
    *   `HomeViewModel` now uses `ObservePopularSeriesUseCase` to get popular series.
*   **Updated Tests:** All relevant unit tests have been updated to reflect these changes.

These changes improve the clarity and flexibility of series data management, particularly around caching and observing popular content.

skaik-mo committed:
Refactor: Observe recently released series from local data source

This commit refactors the way recently released series are fetched and observed.

Key changes:
- `SeriesDao.getRecentlyReleasedSeries` and `SeriesLocalDataSource.getRecentlyReleasedSeries` now return a `Flow<List<SeriesCacheDto>>` instead of `List<SeriesCacheDto>`.
- `SeriesRepository.observeRecentlyReleased` is introduced to observe recently released series from the local data source. It fetches from remote if the local cache is empty.
- `GetRecentlyReleasedSeriesUseCase` is moved to the `recentlyReleased` package.
- `ObserveRecentlyReleasedSeriesUseCase` is introduced to observe recently released series.
- `HomeViewModel` now uses `ObserveRecentlyReleasedSeriesUseCase` to get recently released series.
- Updated tests to reflect these changes.

skaik-mo committed:
Refactor: Observe top-rated series with Flow

This commit refactors the way top-rated series are observed by:

- Modifying `SeriesDao.getTopRatedSeries` to return a `Flow<List<SeriesCacheDto>>`.
- Updating `SeriesLocalDataSourceImp.getTopRatedSeries` and `SeriesLocalDateSource.getTopRatedSeries` to reflect this change.
- Introducing `SeriesRepository.observeTopRated` which returns a `Flow<List<Series>>` and fetches from remote if the local cache is empty.
- Removing the direct `getTopRated` call from `HomeViewModel` and replacing it with observation of `observeTopRatedSeriesUseCase`.
- Adding `ObserveTopRatedSeriesUseCase` to handle the observation logic.
- Renaming `GetTopRatedSeriesUseCase` and its test file for clarity within the `topRated` package.
- Updating tests to reflect the new Flow-based approach.

skaik-mo committed:
Refactor: Make `getMatchesYourVibe` in `SeriesLocalDataSourceImp` return Flow

This commit refactors the `getMatchesYourVibe` function in `SeriesLocalDataSourceImp` to return a `Flow` instead of a suspend function. This change propagates through the repository and use cases, leading to the creation of a new `ObserveMatchesYourVibeSeriesUseCase` to handle the Flow-based observation.

The following changes were made:
- Modified `SeriesLocalDataSourceImp.getMatchesYourVibe` to return `Flow<List<SeriesCacheDto>>`.
- Updated `SeriesDao.getMatchesYourVibeSeries` to return `Flow<List<SeriesCacheDto>>`.
- Updated `SeriesRepositoryImpl` to include `observeMatchesYourVibe` which returns a `Flow` and modified the existing `getMatchesYourVibe` to fetch from remote if local is empty.
- Created `ObserveMatchesYourVibeSeriesUseCase` to expose the Flow of "Matches Your Vibe" series.
- Renamed `GetMatchesYourVibeSeriesUseCase` and its test file to be under a `matchesYourVibe` package.
- Updated `HomeViewModel` to use `ObserveMatchesYourVibeSeriesUseCase`.
- Updated relevant test files to reflect these changes.

Mostafa-khaled21 commented on datasource/local/media/src/main/java/com/giraffe/media/movie/MovieLocalDataSourceImpl.kt:
```diff
@@ -17,19 +18,29 @@ import javax.inject.Inject
 class MovieLocalDataSourceImpl @Inject constructor(
     private val movieDao: MovieDao
 ) : MoviesLocalDataSource {
+    override suspend fun addMovie(movie: MovieCacheDto) {
```
why did't use safecall?

Mostafa-khaled21 commented on domain/media/src/main/java/com/giraffe/media/movie/usecase/genre/SyncMoviesGenresUseCase.kt:
```diff
@@ -1,12 +1,11 @@
 import javax.inject.Inject
 
-class GetMoviesGenresUseCase @Inject constructor(
+class SyncMoviesGenresUseCase @Inject constructor(
```
Why is Sync and there is no logic here Make sure it is remote or local here? I think the repo is sync

Mostafa-khaled21 reviewed (dismissed):
I think the name Observe is in the case where the data is coming and not stopping at all, but in a case like match vibs we have a limit. So why observe and not get?

mohamedebrahem13 commented on datasource/local/media/src/main/java/com/giraffe/media/series/SeriesLocalDataSourceImp.kt:
```diff
@@ -105,7 +100,30 @@ class SeriesLocalDataSourceImp @Inject constructor(
+    }
+
+
+    override suspend fun clearExceptRecentlyViewed() {
```
The name is not convenient here you said that you clear recently viewed but the function act deffent.

mohamedebrahem13 commented on datasource/local/media/src/test/java/com/giraffe/media/series/SeriesLocalDataSourceImpTest.kt:
```diff
@@ -218,6 +214,34 @@ class SeriesLocalDataSourceImpTest {
+    }
+
+    @Test
+    fun `clearAll clears DAO recent table`() = runTest {
```
After clear try to add test case to get data that you just delete and make sure that the data is cleared not only to check if the function is called or not

mohamedebrahem13 commented on domain/media/src/main/java/com/giraffe/media/series/usecase/ClearSeriesCacheUseCase.kt:
```diff
@@ -6,5 +6,14 @@ import javax.inject.Inject
+    suspend fun clearAll() =
+        seriesRepository.clearAll()
+
+    suspend fun clearExceptRecentlyViewed() =
```
Can you explain the difference between both except and normal clear

skaik-mo replied:
`clearAll` should clear all series related DAOs (series, genres, top rated, popular, etc.).
`clearExceptRecentlyViewed` should clear everything except recently viewed and genres, including the tables for Popular, Recently Released, Top Rated, and Matches Your Vibe.

mohamedebrahem13 commented on domain/media/src/main/java/com/giraffe/media/series/usecase/GetSeriesByNameUseCase.kt:
```diff
@@ -8,10 +8,9 @@ class GetSeriesByNameUseCase @Inject constructor(
-        return topGenre?.let { genre ->
-            results.sortedByDescending { it.genreIDs.contains(genre.id) }
+        val results = seriesRepository.getByName(name = seriesName, page = page)
+        return seriesRepository.getTopGenreCount()?.let { genre ->
```
You can extract this with private helper fun in use case for more readability

Mostafa-khaled21 commented on domain/media/src/main/java/com/giraffe/media/movie/usecase/recentlyViewed/SyncRecentlyViewedMoviesUseCase.kt:
```diff
@@ -0,0 +1,11 @@
+import com.giraffe.media.movie.repository.MovieRepository
+import javax.inject.Inject
+
+class SyncRecentlyViewedMoviesUseCase @Inject constructor(
```
same

Mostafa-khaled21 commented on presentation/explore/src/main/java/com/giraffe/presentation/explore/screen/search/SearchViewModel.kt:
```diff
@@ -116,7 +116,7 @@ class SearchViewModel @Inject constructor(
 
     override fun clearAllRecentViewedPosters() {
         safeExecute {
-            val job1 = launch { clearRecentlyViewedSeriesUseCase() }
```
why job ? Why not coroutineScope?

Mostafa-khaled21 commented on presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/MatchesYourVibesStrategy.kt:
```diff
@@ -17,7 +17,7 @@ class MatchesYourVibesStrategy(
     override suspend fun loadData(page: Int, pageSize: Int): List<PosterMedia> {
 
-        val matchesYourVibeMovies = getMatchesYourVibeMovies(page = page, limit = pageSize)
+        val matchesYourVibeMovies = getMatchesYourVibeMovies.invoke(page = page, limit = pageSize)
```
why use invoke  ?       

 val matchesYourVibeMovies = getMatchesYourVibeMovies(page = page, limit = pageSize)

Mostafa-khaled21 commented on presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/strategy/RecentlyViewedStrategy.kt:
```diff
@@ -1,23 +1,23 @@
 ) : CategoryMediaStrategy {
     override suspend fun loadData(page: Int, pageSize: Int): List<PosterMedia> {
-        val recentMovies = getRecentlyViewedMovies(page, pageSize).first()
+        val recentMovies = observeRecentlyViewedMoviesUseCase.invoke(page, pageSize).first()
```
same

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  fix: GetCollectionsUseCaseTest
  save bottom bar state on navigation
  refactor: use string endpoints instead of companion object rename function to setUserAsNotGuest rename function to setUserAsGuest
  fix: issue when login as a guest then login as a user
  Set `autoMirrored` to true for outline and duotone icons
  Refactor: Replace IsLoggedInUseCase with IsLoggedInByAccountUseCase
  handle get movie or series wen user is guest
  Show loading state on guest login
  Implement login as guest and account login check use cases
  Add guest user flag to AuthenticationDatastore
  Allow users to log in as guests
  Refactor: Extract URL constants in UserApiService
  Add guest user functionality to local data source
  remove unused class
  remove unused import

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/seriesdetails/SeriesDetailsViewModel.kt
#	presentation/match/src/main/java/com/giraffe/match/screen/result/MatchResultViewModel.kt

skaik-mo committed:
Refactor: Sync recently viewed series on settings update

This commit introduces the ability to synchronize recently viewed series data when settings are updated.

Key changes include:
- Added `SyncRecentlyViewedSeriesUseCase` and its test.
- Implemented `syncRecentlyViewedSeries` in `SeriesRepositoryImpl` to fetch and cache details for recently viewed series.
- Added `addSeries` and `getAllRecentlyViewedSeries` to `SeriesDao` and `SeriesLocalDataSource`.
- Renamed `GetRecentlyViewedSeriesUseCase` to `ObserveRecentlyViewedSeriesUseCase` and updated its usages.
- Integrated `syncRecentlyViewedSeriesUseCase` into `SettingsViewModel` to trigger synchronization when settings are changed.
- Updated mappers for `SeriesDetailsDto` to `SeriesCacheDto`.

skaik-mo committed:
Fix: Prioritize new items in recently viewed lists

This commit ensures that newly viewed movies and series appear at the top of the "Recently Viewed" lists in both the Home and Explore screens.

- In `RecentlyViewedStrategy.kt`, the filtering for `recentViewedAt != null` was removed as items fetched are inherently recently viewed.
- In `SearchViewModel.kt`, the order of concatenation was reversed when updating `recentPosters` for both movies and series, placing new items first.

skaik-mo committed:
Refactor: Update recently viewed series handling

This commit refactors how recently viewed series are handled.

Key changes:
- `SeriesCacheDto` no longer stores `recentViewedAt`. This information is now retrieved directly from the `RECENT_VIEWED_SERIES_TABLE` via a join in `SeriesDao`.
- A new data class `SeriesWithRecentlyViewedAt` is introduced to represent the result of this join.
- Mappers and use cases are updated to use `SeriesWithRecentlyViewedAt` for recently viewed series.
- UI components now sort recently viewed items by `recentViewedAt` in descending order.
- The `Poster` UI model across different presentation modules (`explore`, `home`, `profile`) now includes a `recentViewedAt` field (of type `Long`) to facilitate this sorting.
- Removed the `syncRecentViewedTime` function from `SeriesDao` as it's no longer needed.
- `SeriesLocalDataSource` and `SeriesRepository` now return `SeriesWithRecentlyViewedAt` for recently viewed series and retrieve only IDs for syncing.

skaik-mo committed:
Refactor: Change `recentViewedAt` type from `ULong` to `Long`

The `recentViewedAt` property in `Series` and related test data creators has been changed from `ULong` to `Long`.

This change also involved updating tests for `SeriesLocalDataSourceImpTest` and `SeriesRepositoryImplTest` to reflect the new type and ensure correct data handling, particularly for recently viewed series. The tests now correctly handle `SeriesWithRecentlyViewedAt` which encapsulates the `SeriesCacheDto` and its associated `recentViewedAt` timestamp.

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop: (37 commits)
  Fix poster genres in history and discover screens
  Refactor: Localize date, duration, and rating formats
  remove unused file
  Fix profile data refresh
  Update CODEOWNERS
  Create CODEOWNERS
  delete use case and inject only repo
  make refresh user data logic in repo
  resolve conflict
  delete check condition from repo
  enhance settings view model
  update settings screen to improve code
  update use case to be more simple
  use save collect with observe user profile
  resolve problem with no internet
  add progress bar in screen when user logout
  update state for logout in view model
  add state for logout
  delete unused code
  update get collections use case test
  ...

# Conflicts:
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/util/mapper.kt
#	presentation/profile/src/main/java/com/giraffe/presentation/profile/screens/settings/SettingsViewModel.kt

skaik-mo committed:
Refactor: Rename `clearExceptRecentlyViewed` to `clearAllSeriesExceptRecentlyViewed`

This commit renames the function `clearExceptRecentlyViewed` to `clearAllSeriesExceptRecentlyViewed` across various files for better clarity on its functionality.

The change affects the following:
- `SeriesLocalDataSourceImp.kt`: Updated function name.
- `SeriesRepositoryImpl.kt`: Updated function name.
- `SeriesRepository.kt`: Updated function name in the interface.
- `ClearSeriesCacheUseCase.kt`: Updated function call.
- `SeriesLocalDateSource.kt`: Updated function name in the interface.

Corresponding tests have also been updated:
- `SeriesLocalDataSourceImpTest.kt`: Updated function call and added a new test case for `clearAll` to ensure all data is cleared as expected.
- `SeriesRepositoryImplTest.kt`: Updated function call in test.
- `ClearSeriesCacheUseCaseTest.kt`: Updated function call in test.

skaik-mo committed:
Refactor: Improve code style and clarity in media data sources

This commit refactors the media data sources (local and remote for series, local for movies) to enhance code style and readability.

Key changes:
- Utilized `safeCall` more consistently for concise error handling in local data sources.
- Simplified conditional logic, for example, by returning early if a list is empty in `incrementInteractionCountForGenres`.
- Improved function signatures and return types for clarity in remote and repository layers.
- Made minor adjustments to wrapping and spacing for better code formatting.

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  fix details animation progress lost issue
  fix navigation to cast credit
  add effect error in cast credit
  edit rating format in arabic
  edit date format in arabic
  format date born on actor details
  replace format date form full to short
  remove toast unnecessary and fix release year mapper

skaik-mo committed:
Test: Update CastMember birthday to LocalDate

skaik-mo committed:
Refactor: Optimize recently viewed movies retrieval

- Changed `getAllRecentlyViewedMovies()` to `getRecentlyViewedMovieIds()` to only fetch movie IDs instead of full movie objects.
- Updated related DAO and repository methods to reflect this change.
- This optimization reduces unnecessary data fetching when syncing recently viewed movies.

skaik-mo committed:
Refactor: Rename `getAllRecentlyViewedSeriesRecentlyViewedMovieIds` to `getRecentlyViewedMovieIds`

HendSayed25 commented on repository/media/src/main/java/com/giraffe/media/series/datasource/local/cacheDto/RecentViewedSeriesCacheDto.kt:
```diff
@@ -9,4 +10,9 @@ data class RecentViewedSeriesCacheDto(
+)
+
+data class SeriesWithRecentlyViewedAt(
+    @Embedded val series: SeriesCacheDto,
```
what's the meaning of Embedded ?

HendSayed25 commented on domain/media/src/test/kotlin/com/giraffe/media/series/usecase/topRated/ObserveTopRatedSeriesUseCaseTest.kt:
```diff
@@ -0,0 +1,31 @@
+
+
+    @Test
+    fun `given top rated series, when invoke is called, then return series list`() = runTest {
```
```suggestion
    fun `should return list of top rated series when observeTopRated function in repository called'
```
I think this name is more clear

HendSayed25 commented on domain/media/src/test/kotlin/com/giraffe/media/series/usecase/recentlyReleased/ObserveRecentlyReleasedSeriesUseCaseTest.kt:
```diff
@@ -0,0 +1,30 @@
+        ObserveRecentlyReleasedSeriesUseCase(repository)
+
+    @Test
+    fun `given recently released series, when invoke is called, then return series list`() =
```
the same here :
```suggestion
    fun `should return list of recently released series when the usecase is called  `() =
```

HendSayed25 commented on domain/media/src/main/java/com/giraffe/media/series/usecase/genre/GetSeriesGenresByIdsUseCase.kt:
```diff
@@ -1,10 +1,12 @@
 ) {
-    suspend operator fun invoke(genreIDs: List<Int>) = seriesRepository.getGenresByIds(genreIDs)
+    suspend operator fun invoke(genreIDs: List<Int>) =
+        seriesRepository.getGenresByIds(genreIDs).first()
```
why return only the first ?

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  fix space between date and time
  add on retry click in screens when try to connect to internet again
  handle size of progress icon when add movie or series to collection
  fix date and time issue
  Update README.md

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  fix: test case and show rating
  fix: test case utils in details
  fix: translate unknown to غير معروف
  fix: MediaPoster click shadow
  fix: MediaPoster animation and click shadow
  fix: MediaPoster animation
  fix clip poster and rename to media poster

# Conflicts:
#	domain/media/src/test/kotlin/com/giraffe/media/mediaMember/util/TestDummyData.kt

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  add safe call
  Update presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/CategoryMediaViewModel.kt
  Update presentation/home/src/main/java/com/giraffe/presentation/home/screen/categoryMedia/CategoryMediaViewModel.kt
  Update presentation/profile/src/main/java/com/giraffe/presentation/profile/screens/mycollections/MyCollectionsViewModel.kt
  Update presentation/profile/src/main/java/com/giraffe/presentation/profile/screens/mycollections/MyCollectionsViewModel.kt
  fix collections issues

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  optimize imports
  fix delete session id request

skaik-mo committed:
Refactor: Consolidate cache clearing operations in SettingsViewModel

The `clearMoviesCacheUseCase`, `clearSeriesCacheUseCase`, and `clearCollectionsCacheUseCase` are now called within a single `safeExecute` block in the `onConfirmLogoutSuccess` function. Previously, `clearCollectionsCacheUseCase` was called in `confirmLogout` and the other two in separate `safeExecute` blocks in `onConfirmLogoutSuccess`.

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  feat: Animate scrolling to the end of the match pager , automatically scrolls to the end of the `MatchPagerScreen` when new content is loaded.

skaik-mo committed:
Update domain/media/src/test/kotlin/com/giraffe/media/series/usecase/topRated/ObserveTopRatedSeriesUseCaseTest.kt

Co-authored-by: Hend Sayed <hendsayed853@gmail.com>

skaik-mo committed:
Update domain/media/src/test/kotlin/com/giraffe/media/series/usecase/recentlyReleased/ObserveRecentlyReleasedSeriesUseCaseTest.kt

Co-authored-by: Hend Sayed <hendsayed853@gmail.com>

skaik-mo committed:
Fix: Correct ObserveTopRatedSeriesUseCaseTest function signature

skaik-mo committed:
Merge branch 'develop' into fix/series-cache-language

* develop:
  Remove "New Collection" button from "Add to Collection" bottom sheet
  Remove unnecessary arguments from AddToCollectionBottomSheetContent
  Translate "You gave it" string
  Refactor: Pass NavController to profile navigation routes
